config_quick_debug.yml

```diff
@@ -1,11 +1,12 @@
 experiment_name: debug_config
 n_repeats: 1
 scenario_params_list:
- - partners_count: 3
-   amounts_per_partner: [0.45, 0.45, 0.1]
+ - partners_count: 4
+   amounts_per_partner: [0.4, 0.4, 0.1, 0.1]
    samples_split_option: 'stratified'
+   advanced_split: [[7, 'shared'], [6, 'shared'], [2, 'specific'], [1, 'specific']]
```

Comment:
I tested it:
![image](https://user-images.githubusercontent.com/5354301/80090448-d29b4580-855f-11ea-8b8f-68d3f4e3c72a.png)

- partner 0 should be the ones with 7 classes ?
- For the "specific" parners, one should have two classes. This is not the case on the graph.

-

Reply:
What the graph shows is OK:

1. Order of partners: this is because you plotted the partners with their index not their id (`partner.partner_id`). Later in the code the `partners_list` is reordered per `partner_id`.
1. Partner 0 (id #2) having only data from 1 cluster: this is because of the data amounts provided. This is expected (and, I think, the best way to handle weird split options like this one in example)

Reply:
1. OK. I fixed it in the plot function --> https://github.com/SubstraFoundation/distributed-learning-contributivity/commit/181aa68af3d7ac9890f71dedc8750d8fdc46ef47

2. Hum yes this is tricky when we combine it with amount_per_partner ! But actually, I would expect, partner 0 (id `#2`) to still have 2 classes but with 50% less data in each class. So it will respect both constraints (`amount_per_partner` and `split`). What do you think ?

Also we have to keep in mind that MNIST is a perfectly balanced dataset. Do you think this implementation still works on a very imbalanced dataset ?

Reply:
On point n°2 yes this would make sense. I don't remember why but I thought about it and decided it was better to draw from specific classes in sequence. Now that you mention it, I find more logical to do as you describe.

On imbalanced datasets:

- in a multi-partner setting, some datasets might be imbalanced
- in our approach, I think mostly it is for the person configuring the experiment to think about it and configure something that makes sense

Reply:
Done